### ISSUE TITLE
"Connection closed before handshake completed" on cluster.get_client()

### ISSUE THREAD

**Issue by ADBurpitt** [dask/dask-gateway#388]
**What happened**:

On calling `cluster.get_client()` I receive the following error:

```
error                                     Traceback (most recent call last)
/opt/conda/lib/python3.7/site-packages/distributed/comm/core.py in connect(addr, timeout, deserialize, handshake_overrides, **connection_args)
    317         # write, handshake = await asyncio.gather(comm.write(local_info), comm.read())
--> 318         handshake = await asyncio.wait_for(comm.read(), time_left())
    319         await asyncio.wait_for(comm.write(local_info), time_left())

/opt/conda/lib/python3.7/asyncio/tasks.py in wait_for(fut, timeout, loop)
    441         if fut.done():
--> 442             return fut.result()
    443         else:

/opt/conda/lib/python3.7/site-packages/distributed/comm/tcp.py in read(self, deserializers)
    215             try:
--> 216                 frames = unpack_frames(frames)
    217 

/opt/conda/lib/python3.7/site-packages/distributed/protocol/utils.py in unpack_frames(b)
     69 
---> 70     (n_frames,) = struct.unpack_from(fmt, b)
     71     lengths = struct.unpack_from(f"{n_frames}{fmt}", b, fmt_size)

error: unpack_from requires a buffer of at least 8 bytes

The above exception was the direct cause of the following exception:

OSError                                   Traceback (most recent call last)
<ipython-input-39-affca45186d3> in <module>
----> 1 client = cluster.get_client()

/opt/conda/lib/python3.7/site-packages/dask_gateway/client.py in get_client(self, set_as_default)
   1081             set_as_default=set_as_default,
   1082             asynchronous=self.asynchronous,
-> 1083             loop=self.loop,
   1084         )
   1085         if not self.asynchronous:

/opt/conda/lib/python3.7/site-packages/distributed/client.py in __init__(self, address, loop, timeout, set_as_default, scheduler_file, security, asynchronous, name, heartbeat_interval, serializers, deserializers, extensions, direct_to_workers, connection_limit, **kwargs)
    742             ext(self)
    743 
--> 744         self.start(timeout=timeout)
    745         Client._instances.add(self)
    746 

/opt/conda/lib/python3.7/site-packages/distributed/client.py in start(self, **kwargs)
    958             self._started = asyncio.ensure_future(self._start(**kwargs))
    959         else:
--> 960             sync(self.loop, self._start, **kwargs)
    961 
    962     def __await__(self):

/opt/conda/lib/python3.7/site-packages/distributed/utils.py in sync(loop, func, callback_timeout, *args, **kwargs)
    351     if error[0]:
    352         typ, exc, tb = error[0]
--> 353         raise exc.with_traceback(tb)
    354     else:
    355         return result[0]

/opt/conda/lib/python3.7/site-packages/distributed/utils.py in f()
    334             if callback_timeout is not None:
    335                 future = asyncio.wait_for(future, callback_timeout)
--> 336             result[0] = yield future
    337         except Exception as exc:
    338             error[0] = sys.exc_info()

/opt/conda/lib/python3.7/site-packages/tornado/gen.py in run(self)
    733 
    734                     try:
--> 735                         value = future.result()
    736                     except Exception:
    737                         exc_info = sys.exc_info()

/opt/conda/lib/python3.7/site-packages/distributed/client.py in _start(self, timeout, **kwargs)
   1048 
   1049         try:
-> 1050             await self._ensure_connected(timeout=timeout)
   1051         except (OSError, ImportError):
   1052             await self._close()

/opt/conda/lib/python3.7/site-packages/distributed/client.py in _ensure_connected(self, timeout)
   1106         try:
   1107             comm = await connect(
-> 1108                 self.scheduler.address, timeout=timeout, **self.connection_args
   1109             )
   1110             comm.name = "Client->Scheduler"

/opt/conda/lib/python3.7/site-packages/distributed/comm/core.py in connect(addr, timeout, deserialize, handshake_overrides, **connection_args)
    323         raise IOError(
    324             f"Timed out during handshake while connecting to {addr} after {timeout} s"
--> 325         ) from exc
    326 
    327     comm.remote_info = handshake

OSError: Timed out during handshake while connecting to gateway://gateway.degreed.com:443/daskgateway.f862f00c39b44c9dadab5134080fedbb after 10 s
```

and the scheduler logs: `distributed.comm.tcp - INFO - Connection closed before handshake completed`

**What you expected to happen**:

The client to connect correctly.

**Minimal Complete Verifiable Example**:

```python
cluster.get_client()
```

**Anything else we need to know?**:

I've been attempting to deploy dask gateway on AKS and this is as far as I've gotten. The deployment is in a cluster with an existing traefik instance being used as our ingress-controller for all our services, and I'm not sure how this affects the dask setup.

In #386 @jacobtomlinson mentioned that the dask deployment doesn't use traefik as an ingress. What does this mean for those of us that have traefik set up elsewhere as our ingress controller?

Do we not have to expose the dask-traefik service?
Can we set the traefik service type to ClusterIP so we don't have an unused LoadBalancer running?
And If this is the case how does dask make use of this traefik instance?

I've exposed the gateway API using the existing traefik instance, so I can create clusters via the API, but connecting to the schedulers isn't working.

**Environment**:

- Dask version: 0.9.0
- Python version:
- Operating System:
- Install method (conda, pip, source): helm

**Comment by Id2ndR** [comment]
I encountered the same behavior.

I solved it by using the same dask version (2.30.0) on my scheduler/worker and my client like suggested [here](https://github.com/dask/dask-gateway/issues/316#issuecomment-702947730).
Before solving it, I used dask==2021.3.0 on the client whereas dask==2.30.0 was provided on the scheduler/worker by the daskhub helm chart.

**Comment by ADBurpitt** [comment]
Thanks, problem solved!
